tests/integ/modin/series/test_between.py

```diff
@@ -0,0 +1,71 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+import datetime as dt
+
+import pandas as native_pd
+import pytest
+
+import modin.pandas as pd
+import snowflake.snowpark.modin.plugin  # noqa: F401
+from snowflake.snowpark.exceptions import SnowparkSQLException
+from tests.integ.modin.utils import eval_snowpark_pandas_result, create_test_series
+from tests.integ.utils.sql_counter import sql_count_checker
+
+
+@sql_count_checker(query_count=1)
+def test_series_between_default_inclusive():
+    eval_snowpark_pandas_result(
+        *create_test_series(list(range(0, 10))), lambda ser: ser.between(2, 8)
+    )
+
+
+# tuples of (data, low, high)
+BETWEEN_TEST_ARGUMENTS = [
```

Comment:
Can we include nulls in the test arguments?

Reply:
I just added, some and it looks like we have some inconsistencies with pandas because we have different behavior because comparing with a NULL value in snowflake always produces a NULL result:

```
>>> pd.Series([0.8, np.nan]) < 1  # snowpark pandas
0    True
1    None
dtype: object
>>> native_pd.Series([0.8, np.nan]) < 1
>>> native_pd.Series([0.8, np.nan]) < np.nan
0    False
1    False
dtype: bool
```
This behavior propagates to `between`:
```
>>> pd.Series([0.8, np.nan]).between(np.nan, 1)
0    None
1    None
dtype: object
>>> native_pd.Series([0.8, np.nan]).between(np.nan, 1)
0    False
1    False
dtype: bool
```

We could get around this by overriding the upstream method and just sticking an `astype(bool)` to coerce the `None` values, but this would break our principle of adhering to SQL behavior. Alternatively, we could just modify the tests to expect None results as needed. What do you think?

Reply:
Let's override the upstream method. Sticking to pandas behavior is the North Star. Snowflake sql behavior is the second best.

Reply:
Should we just override our implementations for the comparison (`<`/`<=` etc.) operators then? Those diverge from pandas (as the code example shows), and fixing those would fix `between` for free.

Reply:
On second thoughts, I think we should stay consistent with the other comparison operators. But let's still have tests covering nulls. You can check the tests used for the binary comparison ops.

Reply:
From what I could tell, the tests in `tests/integ/modin/binary/test_binary_op.py` actually avoided testing comparison operators on data with NULLs. I added some tests for `between` with NULL data, and used `astype(bool)` to cast NULL values at the end.

Reply:
Hmm, I thought they used `_gen_random_int_list_with_nones` for testing with NULLs. Also, I think this is where NULL comparison is handled in tests: https://github.com/snowflakedb/snowpark-python/blob/3c38e82a288e1c47f9034d381cbc1c175b520d31/tests/integ/modin/utils.py#L275

Reply:
All the tests I saw that used `_gen_random_int_list_with_nones` only used arithmetic operators, and skipped comparisons, like this one: https://github.com/snowflakedb/snowpark-python/blob/8e614ee5a4114c25857ac7d5cc56eb0ec439431a/tests/integ/modin/binary/test_binary_op.py#L662

Reply:
Does this one use it?
https://github.com/snowflakedb/snowpark-python/blob/8e614ee5a4114c25857ac7d5cc56eb0ec439431a/tests/integ/modin/binary/test_binary_op.py#L1924

Reply:
Ah, I didn't see this test. It seems like these cast the native pandas data to "Float64" (capital F), which has the same behavior of `None < None` evaluating to `pd.NA`. The tests I added have mixed object + datetime data so we can't always do the same cast here.

Reply:
That's fair. I just wanted to make sure that you're aware of those tests. I think we already have good coverage here.